cfme/tests/infrastructure/test_relationships.py

```diff
@@ -0,0 +1,80 @@
+from utils import testgen
+import pytest
+
+"""
+This test can run only after overcloud cloud provider created and linked to
+undercloud infra provider, need to compare the cloud providers with the
+results of the relationships
+"""
+
+pytestmark = [pytest.mark.meta(server_roles='+smartproxy +smartstate')]
+
+
+def pytest_generate_tests(metafunc):
```

Comment:
As stated previously, this needs cleaning up, the design of this block is used in multiple places but is rather messy.